diff --git a/platform/sifive/fu540/platform.c b/platform/sifive/fu540/platform.c
index c8ead9d..ca3c60c 100644
--- a/platform/sifive/fu540/platform.c
+++ b/platform/sifive/fu540/platform.c
@@ -218,5 +218,6 @@ const struct sbi_platform platform = {
 	.hart_count		= FU540_HART_COUNT,
 	.hart_stack_size	= FU540_HART_STACK_SIZE,
 	.disabled_hart_mask	= FU540_HARITD_DISABLED,
+	.tlb_range_flush_limit  = 1,
 	.platform_ops_addr	= (unsigned long)&platform_ops
 };
